src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_eckey.c

```diff
@@ -0,0 +1,193 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_eckey.h"
+
+#include <assert.h>
+
+EC_KEY* AndroidCryptoNative_NewEcKey(jobject curveParameters, jobject keyPair)
+{
+    assert(curveParameters);
+    assert(keyPair);
+
+    EC_KEY* keyInfo = malloc(sizeof(EC_KEY));
+    memset(keyInfo, 0, sizeof(EC_KEY));
+    keyInfo->refCount = 1;
+    keyInfo->curveParameters = curveParameters;
+    keyInfo->keyPair = keyPair;
+    return keyInfo;
+}
+
+void AndroidCryptoNative_EcKeyDestroy(EC_KEY* r)
+{
+    if (r)
+    {
+        r->refCount--;
+        if (r->refCount == 0)
```

Comment:
We probably want to care about thread safety for refcounting, especially depending on how things like cert.GetECDsaPrivateKey() get implemented (if 20 different managed objects are all tracking that they're backed by the same `EC_KEY*` then there's no way the callers would know that their 20 parallel Dispose() calls aren't thread-kosher)